### PR TITLE
[CI/Build] Add sphinx/rst linter for docs

### PR DESCRIPTION
**Pull request by rafvasq**
Currently there's no linter/style checker for the docs, this PR offers [sphinx-lint](https://github.com/sphinx-contrib/sphinx-lint) to clean up stylistic and formatting errors that don't show up in the build.

### CODE REVIEW

**Review by russellb — COMMENTED** [review]
I took a look at this same linter earlier this week. I didn't post it because it didn't seem to find anything interesting. All I had to disable to get it to run cleanly was:

```sh
sphinx-lint --disable trailing-whitespace,missing-final-newline docs
```

and neither of those warnings seem worth enforcing. My conclusion was that the most important issues are already caught as part of the docs build, which already happens on every PR.

In other words, this didn't seem worth it to me. Did you find anything I missed?

**Comment by rafvasq** [comment]
That's fair, I totally understand if those particular warnings aren't worth enforcing. A few things I'd advocate as helpful for future documentation changes though are the specific the backticks, spaces, underscores needed by things like hyperlinks and roles which don't break the doc build. You'd have to wait for the doc build to complete and only catch these during review.

```
`GPTQ <https://arxiv.org/abs/2210.17323>`_
```
```
:class:`~vllm.LLM`
```

File: .github/workflows/sphinx-lint.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install sphinx-lint
```
Comment (russellb) [inline .github/workflows/sphinx-lint.yml:28]: if you pull in my commit, I would change this to:

```suggestion
          pip install -r requirements-lint.txt
```

File: .github/workflows/sphinx-lint.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Linting docs
+        run: sphinx-lint docs
```
Comment (russellb) [inline .github/workflows/sphinx-lint.yml:30]: and I would change this to:
```suggestion
        run: tools/sphinx-lint.sh
```

This ensures that we use the same arguments to the tool for both CI and local usage.

**Review by russellb — CHANGES_REQUESTED** [review]
In other words, it's quicker to find issues using this during the dev (writing) phase than doing full doc builds? I'd agree with that. To make that a little easier, I'd include it in `format.sh`, as well. I have that in a branch if you want to cherry-pick it to your branch.

https://github.com/vllm-project/vllm/compare/main...russellb:vllm:rst-linting?expand=1

**Comment by russellb** [comment]
> In other words, it's quicker to find issues using this during the dev (writing) phase than doing full doc builds? I'd agree with that. To make that a little easier, I'd include it in `format.sh`, as well. I have that in a branch if you want to cherry-pick it to your branch.
> 
> https://github.com/vllm-project/vllm/compare/main...russellb:vllm:rst-linting?expand=1

btw, I'm also happy to do the opposite - pull your commit into my branch and make the suggested updates. You will still get co-author credit either way. Just let me know how I can help!

**Review by russellb — CHANGES_REQUESTED** [review]
thanks for the updates! one minor comment

File: .github/workflows/sphinx-lint.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "docs/**"
+  pull_request:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (rafvasq) [reply .github/workflows/sphinx-lint.yml:11]: No reason, nice catch!

**Comment by rafvasq** [comment]
@russellb, thanks for the offer to help (and the reviews), I have to get my git practice in whenever I can 😅 I think I got this one right.

**Review by russellb — APPROVED** [review]
lgtm, thanks!

**Review by DarkLight1337 — APPROVED** [review]
Thanks for improving the quality of docs!

**Comment by DarkLight1337** [comment]
Please merge from main to fix the test failures.

**Pull request merged**